test/test_nested_tensor_masking.py

```diff
@@ -3,42 +3,309 @@
 import pdb
 import sys
 import torch
-import nestedtensor
+import nestedtensor as NT
 import unittest
 from utils import TestCase
 import random
 
 import utils
 
 class TestTensorMask(TestCase):
-    # TODO: test comparison operator fails if nested_size doesn't agree.
+    def test_gen_nested_tensor(self):
+        nt = utils.gen_nested_tensor(seed=1, nested_dim=1, tensor_dim=1, size_low=2, size_high=2)
```

Comment:
For these kind of very small cases it would make sense to hardcode the values. It's more readable and there isn't much value for this kind of very basic and pedantic test.

Reply:
Also, you can get more variation. gen_nested_tensor in essence has value for testing invariants such as f(nt(t)) == nt(f(t)) for certain types of functions f.

Reply:
what values would you hardcode here? we are testing if helper `gen_nested_tensor` works as expected.

Reply:
That sounds like it should be a separate test specific to gen_nested_tensor